twill-api/src/main/java/org/apache/twill/api/EventHandler.java

```diff
@@ -124,6 +124,74 @@ public void initialize(EventHandlerContext context) {
   }
 
   /**
+   * Invoked by the application when it starts.
+   *
+   * @param twillAppName name of the current application
+   * @param runId run ID of current application run
+   */
+  public void started(String twillAppName, RunId runId) {
+    // No-op
+  }
+
+  /**
+   * Invoked by the application when new container is launched for {@link TwillRunnable}.
+   *
+   * @param twillAppName name of the current application
+   * @param runId run ID of current application run
+   * @param runnableName name of the runnable to be run in the new container
+   * @param instanceId the instance ID of the runnable instance to be run in the new container
+   * @param containerId the ID of the newly launched container
+   */
+  public void containerLaunched(String twillAppName, RunId runId, String runnableName,
+                                int instanceId, String containerId) {
+    // No-op
+  }
+
+  /**
+   * Invoked by the application when a container is stopped.
+   *
+   * @param twillAppName name of the current application
+   * @param runId run ID of current application run
+   * @param runnableName name of the runnable in the stopped container
+   * @param instanceId the instance ID of the runnable instance run in the stopped container
+   * @param containerId the ID of the stopped container
+   */
+  public void containerStopped(String twillAppName, RunId runId, String runnableName,
+                               int instanceId, String containerId) {
+    // No-op
+  }
+
+  /**
+   * Invoked by the application when all containers complete.
+   *
+   * @param twillAppName name of the current application
+   * @param runId run ID of current application run
+   */
+  public void completed(String twillAppName, RunId runId) {
+    // No-op
+  }
+
+  /**
+   * Invoked by the application when stop command is received to kill the current application.
+   *
+   * @param twillAppName name of the current application
+   * @param runId run ID of current application run
+   */
+  public void killed(String twillAppName, RunId runId) {
+    // No-op
+  }
+
+  /**
+   * Invoked by the application when the application is aborted because of timeout.
+   *
+   * @param twillAppName name of the current application
+   * @param runId run ID of current application run
+   */
+  public void aborted(String twillAppName, RunId runId) {
+    // No-op
+  }
+
+  /**
    * Invoked by the application when shutting down.
    */
   public void destroy() {
```

Comment:
Can't add comment at line 206, but it's better to make the `launchTimeout` method non-abstract.